tests/lib/rules/no-direct-mutation-state.js

```diff
@@ -70,6 +70,14 @@ ruleTester.run('no-direct-mutation-state', rule, {
       '}'
     ].join('\n')
   }, {
+    code: [
+      'class Hello extends React.Component {',
+      '  constructor() {',
+      '    this.state.foo = 1;',
```

Comment:
Assignment doesn't actually matter until we're out of the `constructor` method. That's when React takes over.

But that's also a tangential change to what I wanted to make here - I'm bringing `++` to parity with `=` (you have a test case immediately above this one showing that). If you want to change how you treat assignment as well, that's fine but I think you probably want to do that distinctly.